tests/wpt/web-platform-tests/html/browsers/browsing-the-web/unloading-documents/prompt/004-1.html

```diff
@@ -0,0 +1,21 @@
+<!doctype html>
+004-1
+<script>
+onbeforeunload = function() {
+  onbeforeunload = null;
+  parent.beforeunload_fired = true;
+  setTimeout(function() {
```

Comment:
Can we just call a function on `parent` here that then calls `t.done()`?

Reply:
yes, that is done there https://github.com/servo/servo/pull/20837/files#diff-98df125a78b53a9905ea2e52eb1362d5R4
because here the idea is that the timeout never fires because the document is not salvageable hence will be discarded, and the timeout along with it.